components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java

```diff
@@ -149,6 +150,9 @@ public static boolean isSignatureVerified(SignedJWT signedJWT, String jwksUri) t
             try {
                 return new JWKSBasedJWTValidator().validateSignature(jwtString, jwksUri, alg, MapUtils.EMPTY_MAP);
             } catch (IdentityOAuth2Exception e) {
+                if (e.getCause() != null && e.getCause() instanceof BadJWTException) {
```

Comment:
if it is a expired JWT, then it is a client-side error. We should not log errors for client errors. What is the api response sent to the client/app when they pass the expired cert?